### PR TITLE
Document Data Prepper circuit breakers.

### PR DESCRIPTION
**Pull request by dlvenable**
### Description

Adds documentation on Data Prepper's circuit breakers.

### Issues Resolved

N/A

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _data-prepper/managing-data-prepper/configuring-data-prepper.md [hunk 1]
```diff
@@ -34,40 +38,62 @@ The following section details various configuration options for peer forwarder.
...
+aws_cloud_map_service_name | Conditionally | String | The Cloud Map service name when using AWS Cloud Map service discovery. Required if `discovery_mode` is set to `aws_cloud_map`.
+aws_cloud_map_query_parameters | No | Map | A map of key-value pairs to filter the results based on the custom attributes attached to an instance. Only instances that match all the specified key-value pairs are returned.
+buffer_size | No | Integer | The maximum number of unchecked records the buffer accepts. Number of unchecked records is the sum of the number of records written into the buffer and the num of in-flight records not yet checked by the Checkpointing API. Default is 512.
+batch_size | No | Integer | The maximum number of records the buffer returns on read. Default is 48.
+aws_region | Conditionally | String | The AWS region to use with ACM, S3 or AWS Cloud Map. Required if `use_acm_certificate_for_ssl` is set to true or `ssl_certificate_file` and `ssl_key_file` is AWS S3 path or `discovery_mode` is set to `aws_cloud_map`.
+drain_timeout | No | Duration | The wait time for the peer forwarder to complete processing data before shutdown. Default is 10s.
```
Comment (kolchfa-aws) [inline _data-prepper/managing-data-prepper/configuring-data-prepper.md:55]:
```suggestion
drain_timeout | No | Duration | The wait time for the peer forwarder to complete processing data before shutdown. Default is `10s`.
```

**Review by kolchfa-aws — APPROVED** [review]
LGTM

File: _data-prepper/managing-data-prepper/configuring-data-prepper.md [hunk 2]
```diff
@@ -7,23 +7,27 @@ nav_order: 5
...
+privateKeyPassword | No | String | The password for a private key within keystore. Optional, defaults to empty string.
+serverPort | No | Integer | The port number to use for server APIs. Defaults to 4900.
+metricRegistries | No | List | The metrics registries for publishing the generated metrics. Currently supports Prometheus and CloudWatch. Defaults to Prometheus.
+metricTags | No | Map | A map of key-value pairs as common metric tags to metric registries. The maximum number of pairs is three. Note that `serviceName` is a reserved tag key with `DataPrepper` as default tag value. Alternatively, administrators can set this value through the environment variable `DATAPREPPER_SERVICE_NAME`. If `serviceName` is defined in `metricTags`, that value overwrites those set through the above methods.
+authentication | No | Object | The authentication configuration. Valid option is `http_basic` with `username` and `password` properties. If not defined, the server does not perform authentication.
+processorShutdownTimeout | No | Duration | The time given to processors to clear any in-flight data and gracefully shutdown. Default is 30s.
```
Comment (hdhalter) [inline _data-prepper/managing-data-prepper/configuring-data-prepper.md:27]:
```suggestion
processorShutdownTimeout | No | Duration | The time given to processors to clear any in-flight data and gracefully shut down. Default is 30s.
```

File: _data-prepper/managing-data-prepper/configuring-data-prepper.md [hunk 3]
```diff
@@ -7,23 +7,27 @@ nav_order: 5
...
-sinkShutdownTimeout | No | Duration | Time given to sinks to clear any in-flight data and gracefully shutdown. Default is 30s.
+keyStoreFilePath | No | String | The path to a .jks or .p12 keystore file. Required if `ssl` is true.
+keyStorePassword | No | String | The password for keystore. Optional, defaults to empty string.
+privateKeyPassword | No | String | The password for a private key within keystore. Optional, defaults to empty string.
+serverPort | No | Integer | The port number to use for server APIs. Defaults to 4900.
+metricRegistries | No | List | The metrics registries for publishing the generated metrics. Currently supports Prometheus and CloudWatch. Defaults to Prometheus.
+metricTags | No | Map | A map of key-value pairs as common metric tags to metric registries. The maximum number of pairs is three. Note that `serviceName` is a reserved tag key with `DataPrepper` as default tag value. Alternatively, administrators can set this value through the environment variable `DATAPREPPER_SERVICE_NAME`. If `serviceName` is defined in `metricTags`, that value overwrites those set through the above methods.
```
Comment (hdhalter) [inline _data-prepper/managing-data-prepper/configuring-data-prepper.md:25]:
```suggestion
metricTags | No | Map | A map of key-value pairs as common metric tags to metric registries. The maximum number of pairs is three. Note that `serviceName` is a reserved tag key with `DataPrepper` as the default tag value. Alternatively, administrators can set this value through the environment variable `DATAPREPPER_SERVICE_NAME`. If `serviceName` is defined in `metricTags`, that value overwrites those set through the above methods.
```

File: _data-prepper/managing-data-prepper/configuring-data-prepper.md [hunk 4]
```diff
@@ -7,23 +7,27 @@ nav_order: 5
...
+authentication | No | Object | The authentication configuration. Valid option is `http_basic` with `username` and `password` properties. If not defined, the server does not perform authentication.
+processorShutdownTimeout | No | Duration | The time given to processors to clear any in-flight data and gracefully shutdown. Default is 30s.
+sinkShutdownTimeout | No | Duration | The time given to sinks to clear any in-flight data and gracefully shutdown. Default is 30s.
```
Comment (hdhalter) [inline _data-prepper/managing-data-prepper/configuring-data-prepper.md:28]:
```suggestion
sinkShutdownTimeout | No | Duration | The time given to sinks to clear any in-flight data and gracefully shut down. Default is 30s.
```

File: _data-prepper/managing-data-prepper/configuring-data-prepper.md [hunk 5]
```diff
@@ -7,23 +7,27 @@ nav_order: 5
...
-authentication | No | Object | Authentication configuration. Valid option is `http_basic` with `username` and `password` properties. If not defined, the server does not perform authentication.
-processorShutdownTimeout | No | Duration | Time given to processors to clear any in-flight data and gracefully shutdown. Default is 30s.
-sinkShutdownTimeout | No | Duration | Time given to sinks to clear any in-flight data and gracefully shutdown. Default is 30s.
+keyStoreFilePath | No | String | The path to a .jks or .p12 keystore file. Required if `ssl` is true.
+keyStorePassword | No | String | The password for keystore. Optional, defaults to empty string.
+privateKeyPassword | No | String | The password for a private key within keystore. Optional, defaults to empty string.
+serverPort | No | Integer | The port number to use for server APIs. Defaults to 4900.
+metricRegistries | No | List | The metrics registries for publishing the generated metrics. Currently supports Prometheus and CloudWatch. Defaults to Prometheus.
```
Comment (hdhalter) [inline _data-prepper/managing-data-prepper/configuring-data-prepper.md:24]:
```suggestion
metricRegistries | No | List | The metrics registries for publishing the generated metrics. Currently supports Prometheus and Amazon CloudWatch. Defaults to Prometheus.
```

**Review by hdhalter — APPROVED** [review]
Just a few tweaks.

**Pull request merged**